javascript/jsx/JavaScriptLexer.g4

```diff
@@ -0,0 +1,815 @@
+Var:                            'var';
+Catch:                          'catch';
+Finally:                        'finally';
+Return:                         'return';
+Void:                           'void';
+Continue:                       'continue';
+For:                            'for';
+Switch:                         'switch';
+While:                          'while';
+Debugger:                       'debugger';
+Function:                       'function';
+This:                           'this';
+With:                           'with';
+Default:                        'default';
+If:                             'if';
+Throw:                          'throw';
+Delete:                         'delete';
+In:                             'in';
+Try:                            'try';
+As:                             'as';
+From:                           'from';
+
+/// Future Reserved Words
+
+Class:                          'class';
+Enum:                           'enum';
+Extends:                        'extends';
+Super:                          'super';
+Const:                          'const';
+Export:                         'export';
+Import:                         'import';
+
+Async:                          'async';
+Await:                          'await';
+
+/// The following tokens are also considered to be FutureReservedWords
+/// when parsing strict mode
+
+Implements:                     'implements' {this.IsStrictMode()}?;
+StrictLet:                      'let' {this.IsStrictMode()}?;
+NonStrictLet:                   'let' {!this.IsStrictMode()}?;
+Private:                        'private' {this.IsStrictMode()}?;
+Public:                         'public' {this.IsStrictMode()}?;
+Interface:                      'interface' {this.IsStrictMode()}?;
+Package:                        'package' {this.IsStrictMode()}?;
+Protected:                      'protected' {this.IsStrictMode()}?;
+Static:                         'static' {this.IsStrictMode()}?;
+Yield:                          'yield' {this.IsStrictMode()}?;
+
+/// Identifier Names and Identifiers
+
+Identifier:                     IdentifierStart IdentifierPart*;
+/// String Literals
+StringLiteral:                 ('"' DoubleStringCharacter* '"'
+             |                  '\'' SingleStringCharacter* '\'') {this.ProcessStringLiteral();}
+             ;
+
+// TODO: `${`tmp`}`
+TemplateStringLiteral:          '`' ('\\`' | ~'`')* '`';
+
+WhiteSpaces:                    [\t\u000B\u000C\u0020\u00A0]+ -> channel(HIDDEN);
+
+LineTerminator:                 [\r\n\u2028\u2029] -> channel(HIDDEN);
+
+/// Comments
+
+
+HtmlComment:                    '<!--' .*? '-->' -> channel(HIDDEN);
+CDataComment:                   '<![CDATA[' .*? ']]>' -> channel(HIDDEN);
+UnexpectedCharacter:            . -> channel(ERROR);
+CDATA:                          '<![CDATA[' .*? ']]>' -> channel(HIDDEN);
+
+//
+// html tag declarations
+//
+mode TAG;
+
+TAG_OPEN
```

Comment:
Following other part's naming convention is recommended.

Reply:
fixed.